communication/mavlink_waypoint_handler.cpp

```diff
@@ -819,17 +814,14 @@ mav_result_t Mavlink_waypoint_handler::set_auto_landing(Mavlink_waypoint_handler
 
         //waypoint_handler->navigation_.dubin_state = DUBIN_INIT;
 
-        local_position_t landing_position = waypoint_handler->position_estimation_.local_position;
-        landing_position.pos[Z] = -5.0f;
+        local_position_t landing_position = waypoint_handler->ins_.position_lf();
+        landing_position[Z] = -5.0f;
         if (packet->param1 == 1)
         {
             print_util_dbg_print("Landing at a given location\r\n");
 
-            landing_position.pos[X] = packet->param5;
-            landing_position.pos[Y] = packet->param6;
-            
-            landing_position.heading = waypoint_handler->position_estimation_.local_position.heading;
```

Comment:
Why do you remove it ?
I agree that it seams arbitrary to set it to your current heading, but if you remove it will also be arbitrarily facing north.